regions_www/m/server.R

```diff
@@ -19,804 +19,1071 @@
 # Setup #
 # # # # #
 
-# Colours
-zcols <- "RdYlBu" # for colourbrewer scale (see get_colour_ramp in pct-shiny-funs.R)
-shiny_root <- file.path("..", "..")
-# expect pct-data as a sibling of pct-shiny
-data_dir_root <- file.path(shiny_root, '..', 'pct-data')
-
-# Packages, only reguarly used packages are loaded into the global space
-# the others must be installed but are used with the package prefix, e.g. DT::
+## Functions
+source("pct_shiny_funs.R", local = T)
+
+## Packages (Only regularly used packages are loaded into the global space, the others must be installed but are used with the package prefix, e.g. DT::)
 available_locally_pkgs <- c("shiny", "leaflet", "sp")
 must_be_installed_pkgs <- c("rgdal", "rgeos", "shinyjs", "dplyr", "readr", "geojsonio", "DT")
 
-on_server <- grepl('^/var/shiny/pct-shiny', getwd())
-
-data_sha <- as.character(readLines(file.path(shiny_root, "data_sha")))
-
-# Create a df to store LSOA legend information
-lsoa_legend_df <- data.frame(
-  colours = c("#9C9C9C", "#FFFF73", "#AFFF00", "#00FFFF", "#30B0FF", "#2E5FFF", "#0000FF", "#FF00C5"),
-  labels = c( "1-9", "10-49", "50-99", "100-249", "250-499", "500-999", "1000-1999", "2000+" )
-)
+## Path directories to load data (expect regional data as a sibling of interface_root)
+interface_root <- file.path("..", "..")
+data_regional_root <-  file.path(interface_root, '..', 'pct-outputs-regional')
+data_national_root <-  file.path(interface_root, '..', 'pct-outputs-national')
+
+## Regional sha file [Anna question: next 7 lines added by me, wasn't clear to me otherwise how read the latest master branch? Or was the intention to set manually?]
+gitArgs_reg <- c("--git-dir", file.path(data_regional_root, ".git"), "rev-parse", "--short", "HEAD", ">", file.path(interface_root, "outputs_regional_sha"))
```

Comment:
@AnnaGoodman1 Yes, very much it was meant to be set manually as during development the two weren't always in sync (i.e. when variables were renamed or regions were added one by one we might not want to use the head of the data repository)

Reply:
@nikolai-b , Right, sorry I hadn't appreciated that – feel free to revert it.  Also sorry for not looking at this pull request sooner, wwill now start replying below